src/lib/utils/fiat-estimates/fiat-estimates.ts

```diff
@@ -2,180 +2,113 @@ import { derived, writable, get } from 'svelte/store';
-        newPrices[symbol] = 'unsupported';
-      }
-    }
+  const pricesValue = get(prices);
 
-    return {
-      ...prices,
-      ...newPrices,
-    };
+  // Validate all the addresses are valid ETH addresses
+  addresses.forEach((address) => {
+    assert(utils.isAddress(address), `Invalid address: ${address}`);
   });
 
-  // Filter out all unsupported symbols.
-  symbols = symbols.filter((symbol) => SUPPORTED_SYMBOLS.includes(symbol as SupportedSymbol));
+  // If we're already tracking any of the given addresses, remove them from the list.
+  addresses = addresses.filter(
+    (address) => !Object.keys(pricesValue).includes(ethers.utils.getAddress(address)),
+  );
 
-  // Set values for all the supported symbols in prices store to pending.
-  prices.update((prices) => {
-    const newPrices: Prices = {};
+  // Make all the addresses lowercase
+  addresses = addresses.map((address) => address.toLowerCase());
 
-    for (const symbol of symbols) {
-      newPrices[symbol] = 'pending';
-    }
+  // If the list is empty, we're done.
+  if (addresses.length === 0) return;
 
-    return {
-      ...prices,
-      ...newPrices,
-    };
+  // Update all the tracked addresses to have a price of `pending`.
+  addresses.forEach((address) => {
+    prices.set({
+      ...pricesValue,
+      [ethers.utils.getAddress(address)]: 'pending',
+    });
   });
 
-  if (symbols.length === 0) return;
-
-  connection.send({
-    method: 'SUBSCRIBE',
-    params: symbols.map((symbol) => `${symbol.toLowerCase()}usdt@ticker`),
-    id: 1,
+  // Ensure all addresses are known to the ID map and build a list of IDs.
+  // If the address isn't known, assign it a value of `unsupported`.
+  const ids: DataProviderTokenId[] = [];
+  addresses.forEach((address) => {
+    const id = idMap?.[address];
+
+    if (id) {
+      ids.push(id);
+    } else {
+      prices.update(($prices) => {
+        return {
+          ...$prices,
+          [ethers.utils.getAddress(address)]: 'unsupported',
+        };
+      });
+    }
   });
-}
 
-/** Stop tracking the provided symbols. */
-export async function untrack(symbols: string[]) {
-  assert(connection, 'Socket not initialized');
+  // Request the current prices for all tracked assets from /api/fiat-estimates/price/tokenId1,tokenId2,...
+  const priceRes = await fetch(`/api/fiat-estimates/price/${ids.join(',')}`);
 
-  symbols = symbols.map((symbol) => TOKEN_SUBSTITUTIONS[symbol] || symbol);
+  const parsedRes = z.record(z.string(), z.number()).parse(await priceRes.json());
 
-  connection.send({
-    method: 'UNSUBSCRIBE',
-    params: symbols.map((symbol) => `${symbol.toLowerCase()}usdt@ticker`),
-    id: 1,
+  // Update the prices store with the new prices
+  prices.update(($prices) => {
+    return {
+      ...$prices,
+      ...Object.fromEntries(
+        Object.entries(parsedRes).map(([id, price]) => [
+          utils.getAddress(addresses[ids.indexOf(parseInt(id))].toLowerCase()),
```

Comment:
Just noticed this has a chance to pick the wrong address if there are any unsupported addresses... Will fix!

Reply:
Fixed this 👍